Source/Blazorise/Components/Steps/Steps.razor.cs

```diff
@@ -152,6 +188,11 @@ public string SelectedStep
         /// </summary>
         [Parameter] public EventCallback<string> SelectedStepChanged { get; set; }
 
+        /// <summary>
+        /// Disables navigation by clicking on step.
+        /// </summary>
+        [Parameter] public bool DisableClickNavigation { get; set; }
```

Comment:
What do you think that we make it more dynamic? For example have a delegate that returns true is step is allowed.

**Example:**

```cs
[Parameter] public Func<StepContext, bool> NavigationAllowed { get; set; }
```

where StepContext would be something like

```cs
class StepContext
{
  int CurrentStep { get; set; }
  int NextStep { get; set; } // can be anything, prev or next index
}
```

Reply:
It seems like a good idea. Might be useful to add step name aswell.

class StepContext
{
  int CurrentStepIndex { get; set; }
  int NextStepIndex { get; set; }
  string CurrentStepName { get; set; }
  string NextStepName { get; set; }
}

Reply:
That's good. Better name might be StepChangeContext or StepNavigationContext. I'm open to suggestions.

Reply:
Should indexes be one-based or zero-based and what happens if a step has overridden the index by using the `Index` parameter?

Reply:
Good question. Since we already have DataGrid as one-based, I think we should do the same for the Steps.

Reply:
> what happens if a step has overridden the index by using the `Index` parameter?

The expected override should also be one-based. We should better document this so that the user is not confused.

Reply:
Added `NavigationAllowed` parameter, you can check when you have time.

Reply:
Seems good to me now. Thanks.